jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java

```diff
@@ -1,63 +1,194 @@
+        textFieldPage.textFieldNormal.has().text(randomString);
+        textFieldPage.textFieldNormal.clear();
+        textFieldPage.textFieldNormal.has().text("");
+
+        textFieldPage.textFieldWeight.is().enabled();
+        textFieldPage.textFieldWeight.setValue(randomString);
+        textFieldPage.textFieldWeight.has().text(randomString);
+        textFieldPage.textFieldWeight.clear();
+        textFieldPage.textFieldWeight.has().text("");
+
+        textFieldPage.textFieldAdornmentPassword.is().enabled();
+        textFieldPage.textFieldAdornmentPassword.setValue(randomString);
+        textFieldPage.textFieldAdornmentPassword.has().text(randomString);
+        textFieldPage.textFieldAdornmentPassword.clear();
+        textFieldPage.textFieldAdornmentPassword.has().text("");
+
+        textFieldPage.textFieldAmount.is().enabled();
+        textFieldPage.textFieldAmount.setValue(randomString);
+        textFieldPage.textFieldAmount.has().text(randomString);
+        textFieldPage.textFieldAmount.clear();
+        textFieldPage.textFieldAmount.has().text("");
     }
 
     @Test
-    public void timeTextFieldTest() {
-        inputTextFieldTimePage.open();
-
-        textFieldFrame.textField.setValue("16:00PM");
-        textFieldFrame.textField.has().text("16:00");
-        textFieldFrame.textField.setValue("06:00A");
-        textFieldFrame.textField.has().text("06:00");
-        textFieldFrame.textField.setValue("17:96");
-        textFieldFrame.textField.has().text("17:09");
-        textFieldFrame.textField.setValue("17:75");
-        textFieldFrame.textField.has().text("17:07");
-        textFieldFrame.textField.setValue("17:x5");
-        textFieldFrame.textField.has().text("17:05");
+    public void selectTest() {
+        textFieldPage.open();
+
+        textFieldPage.selectElementField.click();
+        textFieldPage.selectElement.selectItemByText(CurrencyItems.USD.currencyItems);
+        Assert.assertEquals(textFieldPage.selectElementField.getText(), CurrencyItems.USD.currencyItems);
+        textFieldPage.selectElementField.click();
+        textFieldPage.selectElement.selectItemByText(CurrencyItems.EUR.currencyItems);
+        Assert.assertEquals(textFieldPage.selectElementField.getText(), CurrencyItems.EUR.currencyItems);
+        textFieldPage.selectElementField.click();
+        textFieldPage.selectElement.selectItemByText(CurrencyItems.BTC.currencyItems);
+        Assert.assertEquals(textFieldPage.selectElementField.getText(), CurrencyItems.BTC.currencyItems);
+        textFieldPage.selectElementField.click();
+        textFieldPage.selectElement.selectItemByText(CurrencyItems.JPY.currencyItems);
+        Assert.assertEquals(textFieldPage.selectElementField.getText(), CurrencyItems.JPY.currencyItems);
+
+        textFieldPage.selectNativeSelect.select(CurrencyItems.USD.currencyItems);
+        Assert.assertEquals(textFieldPage.selectNativeSelect.getText(), CurrencyItems.USD.currencyItems);
```

Comment:
just try to use JDI Assert with **has()** action

Reply:
try to reduce lines of code using foreach loop using enum you created: enum CurrencyItems

Reply:
Refactored!